### PR TITLE
feat(aggregator): add cache-friendly headers

### LINKED ISSUE #454: Aggregator: add caching-friendly headers

### PR DESCRIPTION
**Pull request by giac-mysten**
Added headers:

- `Cache-control: public, max-age=86400, stale-while-revalidate=3600`
- `ETag: <BLOB_ID>`

Rationale:

Originally I had considered adding `Cache-control: immutable,...`.
This header ensures that the resource is revalidated (queried from the
server) only if it is stale (`max-age` is expired). However, this is
not great for blob invalidation.

Therefore, I opted for not having `immutable`, so the browser checks
if the resource has changed by sending the get request with
the `ETag`. However, this requires the aggregator to respond 
`304 not modified` in case the blob has not been invalidated.

Therefore, this PR currently depends on #422 . Once that is done,
the aggregator can check if there is an `Etag` header in the request,
query the status of the blob, and respond `304 not modified` if it
has not been invalidated.

`stale-while-revalidate` ensures that expired resources can be
revalidated asynchronously.

Closes #454

### CODE REVIEW

File: crates/walrus-service/src/daemon.rs [hunk 1]
```diff
@@ -106,6 +112,22 @@ async fn retrieve_blob<T: Send + Sync>(
             headers.insert(ACCESS_CONTROL_ALLOW_ORIGIN, HeaderValue::from_static("*"));
             // Prevent the browser from trying to guess the MIME type to avoid dangerous inferences.
             headers.insert(X_CONTENT_TYPE_OPTIONS, HeaderValue::from_static("nosniff"));
+            // Insert headers that help caches distribute Walrus blobs.
+            //
+            // Cache for 1 day, and allow refreshig on the client side. Refreshes use the ETag to
```
Comment (gdanezis) [inline crates/walrus-service/src/daemon.rs]: One day may even be on the short side for prod, but fine for test. Should we have this as a config?

Comment (giac-mysten) [reply crates/walrus-service/src/daemon.rs]: Very good point. This PR is parked for the moment, as I wanted to double check all these parameters and headers with PE before merging. Will revisit the default duration with them.

Comment (mlegner) [reply crates/walrus-service/src/daemon.rs]: How about we merge this as it is now and modify it later if needed?

File: crates/walrus-service/src/daemon.rs [hunk 2]
```diff
@@ -116,6 +118,22 @@ async fn retrieve_blob<T: Send + Sync>(
...
+            // check if the content has changed. This allows invalidated blobs to be removed from
+            // caches. `stale-while-revalidate` allows stale content to be served for 1 hour while
+            // the browser tries to validate it (async revalidation).
+            headers.insert(
+                CACHE_CONTROL,
+                HeaderValue::from_static("public, max-age=86400, stale-while-revalidate=3600"),
```
Comment (quepcompute) [inline crates/walrus-service/src/daemon.rs]: @giac-mysten any particular significance to the revalidation period of 1hr?

Comment (giac-mysten) [reply crates/walrus-service/src/daemon.rs]: Sorry @quepcompute , had missed this question. It's just an attempt to strike a balance between cache friendliness and "blockability".

**Pull request merged**